### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by amashutikov**
[DEMO LINK](https://amashutikov.github.io/react_autocomplete/)

### CODE REVIEW

**Comment by vitvadym** [comment]
image
`h1` didn't change to 'No selected person' after clear input

**Review by vitvadym — APPROVED** [review]
Good job 😎

File: src/components/DropDownMenu.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+};
+
+export const DropDownMenu: React.FC<Props> = ({
+  persons,
+  onSelected,
+}) => {
+  return (
+    <div className="dropdown-menu" role="menu">
+      <div className="dropdown-content">
+        {persons.length === 0
```
Comment (dinaminahi) [inline src/components/DropDownMenu.tsx:16]:
```suggestion
        {!persons.length
```

File: src/components/DropDownMenu.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+                <a
+                  href="/"
+                  className={cn({
+                    'has-text-link': person.sex === 'm',
+                    'has-text-danger': person.sex === 'f',
+                  })}
+                  onClick={(event) => {
+                    event.preventDefault();
+                    onSelected(person.slug, person.name);
+                  }}
```
Comment (dinaminahi) [inline src/components/DropDownMenu.tsx:39]: Create a handler for this

File: src/App.tsx [hunk 3]
```diff
@@ -17,40 +54,20 @@ export const App: React.FC = () => {
...
-
-            <div className="dropdown-item">
-              <p className="has-text-danger">Elisabeth Hercke</p>
-            </div>
-          </div>
-        </div>
+        {appliedQuery && (
+          <DropDownMenu
+            persons={visiblePeople}
+            onSelected={handleUserSelect}
```
Comment (dinaminahi) [inline src/App.tsx:68]:
```suggestion
            onPersonSelect={handleUserSelect}
```

File: src/App.tsx [hunk 4]
```diff
@@ -17,40 +54,20 @@ export const App: React.FC = () => {
             type="text"
             placeholder="Enter a part of the name"
             className="input"
+            value={query}
+            onChange={(event) => {
+              applyQuery(event.target.value);
+              handleQueryChange(event.target.value);
+            }}
```
Comment (dinaminahi) [inline src/App.tsx:61]: Create a handler for this

File: src/App.tsx [hunk 5]
```diff
@@ -1,14 +1,51 @@
...
+    debounce(setAppliedQuery, 1000),
+    [],
+  );
+
+  const visiblePeople = useMemo(() => {
+    return peopleFromServer.filter(person => {
+      return person.name
+        .toLowerCase()
+        .includes(appliedQuery
+          .toLowerCase().trim());
```
Comment (dinaminahi) [inline src/App.tsx:22]:
```suggestion
      const name = person.name.toLowerCase();
      const searchedName = appliedQuery.toLowerCase().trim();
      return name.includes(searchedName);
```

**Review by dinaminahi — CHANGES_REQUESTED** [review]
Good job 👍 
Let's do some improvements

**Review by dffUqp — APPROVED** [review]
:fire:

**Pull request closed without merging**